geopandas/array.py

```diff
@@ -407,7 +407,9 @@ def check_geographic_crs(self, stacklevel):
             warnings.warn(
                 "Geometry is in a geographic CRS. Results from "
                 f"'{inspect.stack()[1].function}' are likely incorrect. "
-                "Use 'GeoSeries.to_crs()' to re-project geometries to a "
+                "To obtain more accurate results, you can either enable the "
+                "geodesic calculation option by setting `geopandas.options.geodesic_calculation = True`, "
+                "or use 'GeoSeries.to_crs()' to re-project geometries to a "
```

Comment:
We need to make a distinction for cases where this is true (area and length for now) and for cases where the toggle does nothing, like buffer or centroid.  We might want to add a keyword here and enable the hint to the geodesic calculation only when needed. 

I also think we should expand it in such a case and mention that in GeoPandas 2.0, this will be enabled by default.